kevm-pyk/src/kevm_pyk/__main__.py

```diff
@@ -201,6 +202,69 @@ def wrap_process_pool(workers: int) -> Iterator[ZeroProcessPool | ProcessPool]:
             yield pp
 
 
+class JSONEncoder(json.JSONEncoder):
+    def default(self, obj: Any) -> Any:
+        if isinstance(obj, FrozenDict):
+            return json.JSONEncoder.encode(self, dict(obj))
+        return json.JSONEncoder.default(self, obj)
+
+
+@dataclass
+class KClaimJob:
+    claim: KClaim
+    dependencies: list[KClaimJob]
+
+    @property
+    def digest(self) -> str:
```

Comment:
The `digest` property computes the hash of dependencies every time it's accessed. If the dependencies don't change frequently, maybe you should cache them.

Reply:
It is now a cached property, and to make sure it doesn't change, I've made the class a frozen dataclass.